6. Database/models/product.js

```diff
@@ -0,0 +1,34 @@
+const products = [];
+let id =1;
+
+const db = require('../utils/database');
+
+module.exports = class product {
+    constructor(title,imageUrl,price,description) {
+        this.title = title;
+        this.imageUrl = imageUrl;
+        this.price = price;
+        this.description = description;
+    }
+
+    save() {
+        this.id = product.id();
+        return db.execute('INSERT INTO Products VALUES(?,?,?,?,?)',[this.id,this.title,this.imageUrl,this.price,this.description]);
+    }
+
+    static fetchAll() {
+        return db.execute('SELECT * FROM Products');
+    }
+
+    static delete(index) {
+        products.splice(index,1);
+    }
+
+    static findById(id) {
+        return db.execute('SELECT * FROM Products WHERE Products.id = ? ',[id]);
+    }
+
+    static id() {
+        return id++;
```

Comment:
no need to use id in this way. instead, you can leverage the data types of the database that you are using.

Reply:
actually in the first place i did that , i've set id to primary key and set it to auto increment , so we don't have to add it manually and it will care of itself 
but it was showing error, it was showing that you have to enter id , so that's why i've used this approch , to create the id from the product model and pass it.